### PR TITLE
Add progress observer

### PR DESCRIPTION
**Pull request by ischasny**
Add index progress observer that reports how much the two indexers are lagging behind each other.

### CODE REVIEW

File: progress_observer/internal/client_store.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	dssync "github.com/ipfs/go-datastore/sync"
+	"github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/multicodec"
+	"github.com/ipni/storetheindex/api/v0/ingest/schema"
+
+	// Import so these codecs get registered.
```
Comment (masih) [inline progress_observer/internal/client_store.go:16]: Need these?

File: progress_observer/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	provider := metric.NewMeterProvider(metric.WithReader(m.exporter))
+	meter := provider.Meter("ipni/indexObserver")
+
+	if m.lag, err = meter.Int64Histogram("ipni/indexObserver/lag",
+		instrument.WithUnit(unit.Milliseconds),
+		instrument.WithDescription("Lag in advertisements between target and source indexers")); err != nil {
+		return nil, err
+	}
+
+	if m.count, err = meter.Int64ObservableGauge("ipni/indexObserver/count",
```
Comment (masih) [inline progress_observer/metrics/metrics.go:77]: Use all lower case names. if you need to divide words use `_`.

File: progress_observer/progress_observer.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+var log = logging.Logger("progress_observer")
+
+const (
+	// depth is a depth of chain segment that is fetched form the publisher
+	depth = 1000
+	// parallelism is a max number of concurrent goroutines
+	parallelism = 50
+)
+
+type mismatchInfo struct {
```
Comment (masih) [inline progress_observer/progress_observer.go:28]: `progressInfo`?

**Review by masih — APPROVED** [review]

File: progress_observer/internal/client_store.go (hunk 1 above)
Comment (ischasny) [reply progress_observer/internal/client_store.go:16]: Copy and paste. Removed.

**Pull request merged**